src/python/pants/source/source_root.py

```diff
@@ -17,7 +19,12 @@ class SourceRootCategories:
   ALL = [UNKNOWN, SOURCE, TEST, THIRDPARTY]
 
 
-SourceRoot = namedtuple('_SourceRoot', ['path', 'langs', 'category'])
+class SourceRoot(datatype([('path', str), 'langs', ('category', str)])):
+  pass
+
+
+class AllExistingSourceRoots(Collection.of(SourceRoot)):
```

Comment:
The type should be `SourceRootCollection` or something. "All existing" is a description of what happens to be inside one specific instance we care about, but it's not a property of the type.

Reply:
@stuhood 's argument was that this type should be a newtype wrapping the more-generic `Collection.of(SourceRoot)` to indicate what we're using that collection for in this change (and presumably if we needed to do something else with a `Collection.of(SourceRoot)` somewhere else, we'd create a new wrapper type for *that* purpose).

Reply:
Oh, hm, yeah, I guess the engine requires those to all be separate so it knows what to return.

Let's just name this `AllSourceRoots` though. The "existing" is redundant. The concept of a non-existing source root (i.e., a string that matches the pattern but isn't a directory on disk) is a subtlety not normally needed by users of this code.

Reply:
Fine with that.